pyoozie/builder.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2017 "Shopify inc." All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+from __future__ import unicode_literals
+
+from pyoozie.coordinator import Coordinator
+from pyoozie.tags import Configuration
+
+
+def _workflow_submission_xml(username, workflow_xml_path, configuration=None, indent=False):
+    """Generate a Workflow XML submission message to POST to Oozie."""
+    submission = Configuration(configuration)
+    submission.update({
+        'user.name': username,
+        'oozie.wf.application.path': workflow_xml_path,
+    })
+    return submission.xml(indent)
+
+
+def _coordinator_submission_xml(username, coord_xml_path, configuration=None, indent=False):
+    """Generate a Coordinator XML submission message to POST to Oozie."""
+    submission = Configuration(configuration)
+    submission.update({
+        'user.name': username,
+        'oozie.coord.application.path': coord_xml_path,
+    })
+    return submission.xml(indent)
+
+
+class WorkflowBuilder(object):
+
+    def __init__(self, name):
+        # Initially, let's just use a static template and only one action payload and one action on error
+        self._name = name
+        self._action_name = None
+        self._action_payload = None
+        self._action_error = None
+        self._kill_message = None
+
+    def add_action(self, name, action, action_on_error, kill_on_error='${wf:lastErrorNode()} - ${wf:id()}'):
+        # Today you can't rename your action and you can only have one, but in the future you can add multiple
+        # named actions
+        if any((self._action_name, self._action_payload, self._action_error, self._kill_message)):
```

Comment:
Ha. I considered suggesting this pattern as a replacement for the multiline `if`, but decided not to. I guess we agree 🙂